chromium_src/net/url_request/url_request_http_job.cc

```diff
@@ -13,25 +13,52 @@
 
 namespace {
 
-bool ShouldUseEphemeralStorage(net::URLRequestHttpJob* http_job) {
+bool ShouldUseEphemeralStorage(net::URLRequest* request) {
   if (!base::FeatureList::IsEnabled(net::features::kBraveEphemeralStorage))
     return false;
 
-  const net::IsolationInfo& isolation_info =
-      http_job->request()->isolation_info();
+  const net::IsolationInfo& isolation_info = request->isolation_info();
   if (!isolation_info.top_frame_origin().has_value() ||
       !isolation_info.frame_origin().has_value())
     return false;
   if (*isolation_info.top_frame_origin() == *isolation_info.frame_origin())
     return false;
 
+  if (url::Origin::Create(request->url()) == *isolation_info.top_frame_origin())
+    return false;
+
   return true;
 }
 
 }  // namespace
 
 #define BRAVE_ADDCOOKIEHEADERANDSTART                                          \
-  if (ShouldUseEphemeralStorage(this))                                         \
+  bool use_ephemeral_storage = ShouldUseEphemeralStorage(request_);            \
+  if (use_ephemeral_storage) {                                                 \
+    net::CookieInclusionStatus returned_status;                                \
+    auto cookie = net::CanonicalCookie::Create(                                \
+        request_->url(), "a=a", base::Time::Now(), base::nullopt,              \
+        &returned_status);                                                     \
```

Comment:
This param has a default value of `nullptr` and we don't seem to be using `returned_status` afterwords.